flax/errors.py

```diff
@@ -69,67 +69,82 @@ def __init__(self, message):
+  **PRNGs for Module.init()**
+  
+  In this example, two rngs are used:
 
-        EncoderDecoder(...).init({'params': rng1, 'decode': rng2}, ...)
+  * ``params`` is used for initializing the parameters of the model. This rng
+    is used to initialize the ``some_params`` parameter, and for initializing
+    the weights of the ``Dense`` Module used in ``Bar``.
+  
+  * ``dropout`` is used for the dropout rng that is used in ``Bar``.
 
-     Note that even though they may be used inside submodules, the rngs for the
-     collections should be defined at the top-level. So the ``EncoderDecoder``
-     module above may contain a submodule ``Decoder``, which then uses the 
-     ``decode`` collection. The RNGs will be passed down to submodules
-     automatically.
-  """
-  def __init__(self):
-    super().__init__('First argument passed to an init function should be a '
-                     '`jax.PRNGKey` or a dictionary mapping strings to '
-                     '`jax.PRNGKey`.')
+  So, ``Foo`` is initialized as follows::
+    
+    init_rngs = {'params': random.PRNGKey(0), 'dropout': random.PRNGKey(1)}
+    variables = Foo().init(init_rngs, init_inputs)
 
+  If a Module only requires an rng for ``params``, you can use::
 
-class ApplyScopeInvalidRngsError(FlaxError):
-  """
-  When applying a Module, the `rng` argument should be a dictionary mapping 
-  collections to the PRNGKeys that are used when computing their new values.
+       SomeModule().init(rng, ...)  # Shorthand for {'params': rng}
 
-  For instance, suppose an ``EncoderDecoder`` Module that requires an RNG for
-  decoding tokens based on a categorical probability distribution. Then a 
-  typical call to :meth:`Module.apply() <flax.linen.Module.apply>` looks as
-  follows::
 
-     EncoderDecoder(...).apply(params, ... {'decode': rng2}, ...)
+  **PRNGs for Module.apply()**
+  
+  When applying ``Foo``, only the rng for ``dropout`` is needed, because 
+  ``params`` is only used for initializing the Module parameters::
 
-  Remarks:
+    Foo().apply(variables, inputs, rngs={'dropout': random.PRNGKey(2)})
 
-  * While :meth:`Module.init() <flax.linen.Module.init>` requires a rngs for
-    the collection ``params``, this is not necessary when applying the module,
-    because this collection is only use to initialize the model with.
-  * Even though they may be used inside submodules, the rngs for the collections
-    should be defined at the top-level. So the ``EncoderDecoder`` module above 
-    may contain a submodule ``Decoder``, which then uses the ``decode``
-    collection. The RNGs will be passed down to submodules automatically.
+  If a Module only requires an rng for ``params``, you don't have to provide
+  rngs for apply at all::
+
+       SomeModule().apply(variables, inputs)  # rngs=None
+  """
+  def __init__(self, scope_name, col):
+    super().__init__(f'Scope {scope_name} needs PRNG for collection "{col}"')
+
+
+class BindInvalidRngsError(FlaxError):
```

Comment:
(Like this doesn't give the user much more than `InvalidRngError` -- what is a scope?)